tmt/steps/__init__.py

```diff
@@ -61,24 +59,66 @@ def enabled(self):
         except AttributeError:
             return None
 
+    def status(self, status=None):
+        """
+        Get and set current step status
+
+        The meaning of the status is as follows:
+        todo ... config, data and command line processed (we know what to do)
+        done ... the final result of the step stored to workdir (we are done)
+        """
+        # Update status
+        if status is not None:
+            # Check for valid values
+            if status not in ['todo', 'done']:
+                raise GeneralError(f"Invalid status '{status}'.")
+            # Show status only if changed
+            elif self._status != status:
+                self._status = status
+                self.debug('status', status, color='yellow')
+        # Return status
+        return self._status
+
     def load(self):
-        """ Load step data from the workdir """
-        pass
+        """ Load status and step data from the workdir """
+        try:
+            content = tmt.utils.yaml_to_dict(self.read('step.yaml'))
+            self.debug('Successfully loaded step data.')
+            self.data = content['data']
+            self.status(content['status'])
+        except GeneralError:
+            self.debug('Step data not found.')
```

Comment:
is this correct? Could it be that also the step data is invalid? Maybe we should print it to ease debugging?

Reply:
Catching `GeneralError` here means that data were not found (file could not be read). When you execute a new run with a fresh workdir there is not file do read. I think this should be ok like this.

Reply:
ok, thanks